File: Sources/Olaf/Core/Model/LogLevel.swift
import Foundation

/// Log severity level. Aligned with the `swift-log` and OSLog level model.
public enum LogLevel: Int, Comparable, Sendable, Codable, CaseIterable {
    case trace
    case debug
    case info
    case notice
    case warning
    case error
    case critical

    public static func < (lhs: LogLevel, rhs: LogLevel) -> Bool {
        lhs.rawValue < rhs.rawValue
    }

    /// Stable short name for display/matching (`INFO`, `ERROR`...).
    public var name: String {
        switch self {
        case .trace: return "TRACE"
        case .debug: return "DEBUG"
        case .info: return "INFO"
        case .notice: return "NOTICE"
        case .warning: return "WARNING"
        case .error: return "ERROR"
        case .critical: return "CRITICAL"
        }
    }

    /// Symbol for visually distinguishing the level in the viewer.
    public var symbol: String {
        switch self {
        case .trace: return "🔬"
        case .debug: return "🐞"
        case .info: return "ℹ️"
        case .notice: return "📌"
        case .warning: return "⚠️"
        case .error: return "❌"
        case .critical: return "🔥"
        }
    }
}

File: Android/olaf/src/main/java/com/olaf/LogLevel.kt
package com.olaf

/**
 * Log severity level. Aligned with the iOS package's level model, and with `android.util.Log`
 * on the mirroring side.
 *
 * The ordinal is the on-disk representation (`trace = 0` … `critical = 6`), which is what the
 * iOS package writes into NDJSON too — so history files stay readable across both platforms.
 */
enum class LogLevel {
    TRACE,
    DEBUG,
    INFO,
    NOTICE,
    WARNING,
    ERROR,
    CRITICAL;

    /** Symbol used to tell levels apart at a glance in the viewer. */
    val symbol: String
        get() = when (this) {
            TRACE -> "🔬"
            DEBUG -> "🐞"
            INFO -> "ℹ️"
            NOTICE -> "📌"
            WARNING -> "⚠️"
            ERROR -> "❌"
            CRITICAL -> "🔥"
        }

    companion object {
        /** Reads a level back from its on-disk ordinal; unknown values fall back to [INFO]. */
        fun fromOrdinal(value: Int): LogLevel = entries.getOrElse(value) { INFO }
    }
}
